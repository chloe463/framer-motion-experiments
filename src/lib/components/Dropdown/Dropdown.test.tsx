import { fireEvent, render } from "@testing-library/react";
import { Dropdown } from "./index";

const OPTIONS = [
  "Option1",
  "Option2",
  "Option3",
];

describe("Dropdown", () => {
  it ("should render without crash", () => {
    const { container } = render(<Dropdown options={OPTIONS} itemToString={v => v} value={""} />);
    const input = container.querySelector("div");
    expect(input).toBeTruthy();
  });

  it ("shoud opens options list on click", () => {
    const { container } = render(
      <>
        <Dropdown
          placeholder={"placeholder"}
          options={OPTIONS}
          itemToString={v => v}
          value={""}
          optionsEntryPoingId="options-entry-point"
        />
        <div id="options-entry-point"/>
      </>
    );

    const dropDownBase = container.querySelector("div");
    expect(dropDownBase).toBeTruthy();
    if (!dropDownBase) throw new Error("No dropdown base found");

    fireEvent.click(dropDownBase);
    expect(container).toMatchSnapshot();

    const portalBase = container.querySelector("#options-entry-point");
    expect(portalBase).toBeTruthy();

    if (!portalBase) throw new Error("No portal base found");
    const list = portalBase.querySelector("ul");
    expect(list).toBeTruthy();
    if (!list) throw new Error("No list found");

    expect(list.children.length).toBe(3);
  });
});
