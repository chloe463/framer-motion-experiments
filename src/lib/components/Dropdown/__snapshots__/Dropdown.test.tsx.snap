// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown shoud opens options list on click 1`] = `
<div>
  <div
    class="sc-1e794db1-0 gInAos"
    tabindex="0"
  >
    <label
      class="sc-1e794db1-1 ePUmBU dropdown-placeholder"
    >
      placeholder
    </label>
    <p
      class="sc-1e794db1-2 biYRnI dropdown-selected-value"
    />
    <span
      class="sc-1e794db1-3 hjUGvk dropdown-bottom-border"
    />
  </div>
  <div
    id="options-entry-point"
  >
    <div
      class="sc-8c3d854d-0 iPkWzw"
    >
      <span
        hidden=""
      />
      <ul
        class="sc-fbc7715f-0 fEzBxH"
        style="position: fixed; width: 0px; transform: translate(0px, 0px);"
      >
        <li
          aria-selected="false"
          class="sc-fbc7715f-1 jgLpoS"
          data-selected="false"
          tabindex="0"
        >
          Option1
        </li>
        <li
          aria-selected="false"
          class="sc-fbc7715f-1 jgLpoS"
          data-selected="false"
          tabindex="0"
        >
          Option2
        </li>
        <li
          aria-selected="false"
          class="sc-fbc7715f-1 jgLpoS"
          data-selected="false"
          tabindex="0"
        >
          Option3
        </li>
      </ul>
      <span
        hidden=""
      />
    </div>
  </div>
</div>
`;
